server/api.coffee

```diff
@@ -0,0 +1,29 @@
+apiMethods =
+  '/roomNew': (options) ->
+    try
+      id = Meteor.call 'roomNew',
+        grid: Boolean JSON.parse options.get 'grid'
+      status: 200
+      json:
+        ok: true
+        id: id
+        url: Meteor.absoluteUrl "/r/#{id}"
+    catch e
+      status: 500
+      json:
+        ok: false
+        error: "Error creating new room: #{e}"
+
+WebApp.connectHandlers.use '/api', (req, res, next) ->
+  url = new URL req.url, Meteor.absoluteUrl()
+  if apiMethods.hasOwnProperty url.pathname
+    result = apiMethods[url.pathname] url.searchParams, req, res, next
+    unless res.headersSent
+      res.writeHead result.status, 'Content-type': 'application/json'
+    unless res.writeableEnded
+      res.end JSON.stringify result.json
+  else
+    res.writeHead 404
```

Comment:
Should this also have the `, 'Content-type': 'application/json'`, since the return type here is now JSON, too, after your changes?

I'm not sure whether that actually makes a difference (i.e., whether things like `fetch` or `XMLHttpRequest` care about the `Content-type` header), but it might not be bad to indicate, anyway, that the response contains JSON?

Reply:
Good catch! In fact, given the unification, there's a nice refactor, in 2abe5bf.